util.c

```diff
@@ -623,9 +624,10 @@ static int write_zeroes(int fd, size_t size, off_t offset)
 			return 0;
 	}
 
-	/* Not a regular file, or fallocate not applicable. */
+	/* Not a regular file, non-zero pattern, or fallocate not applicable. */
+	if (size)
+		memset(buf, fillpattern, 4096);
```

Comment:
I don't think that 'if (size) optimization is needed. Anyway, s/4096/sizeof(buf)/ please.

Reply:
fixed.